src/ring_theory/unique_factorization_domain.lean

```diff
@@ -107,6 +107,17 @@ wf_dvd_monoid.induction_on_irreducible a
       by { rw multiset.prod_cons,
            exact hs.2.mul_left _ }⟩⟩)
 
+lemma exists_factors_eq (a : α) (hn0 : a ≠ 0) (hnu : ¬ is_unit a) :
```

Comment:
Do you really need both hypotheses? 0 is not a unit except in the trivial ring, where the statement is vacuously true anyway.

Reply:
Good point, I'll remove the hypothesis.

Reply:
We could also remove `p ≠ 0` from the definition of [prime](https://leanprover-community.github.io/mathlib_docs/algebra/associated.html#prime) then, but that will likely break lots of proofs.

Reply:
Hmm seems we both were confused. If `a` is not a unit then it could still be zero, so `a ≠ 0` is necessary.